packages/flutter_tools/lib/src/build_system/depfile.dart

```diff
@@ -58,6 +58,9 @@ class Depfile {
   ///
   /// If either [inputs] or [outputs] is empty, does not write to the file.
   void writeToFile(File depfile) {
+    if (depfile.existsSync()) {
```

Comment:
The `writeAsStringSync()` below will truncate the file, so I guess this delete is needed for the case where `inputs` or `outputs` is empty. Maybe add a comment to that effect, or move the delete under the if for the inputs/outputs.

Reply:
Moved under the if and added more to the method's doc comment